src/modules/BlockchainApplication/hooks/useBlockchainApplicationManagement.js

```diff
@@ -0,0 +1,74 @@
+import { useCallback, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  addApplicationByChainId as addApplicationAction,
+  deleteApplicationByChainId as deleteApplicationAction
+} from '../store/actions';
+import { selectApplications } from '../store/selectors';
+import { useCurrentBlockchainApplication } from './useCurrentBlockchainApplication';
+import { usePinBlockchainApplication } from './usePinBlockchainApplication';
+import { BLOCKCHAIN_APPLICATIONS_MOCK } from '../mocks';
+
+/**
+ * Hook that handle all the logic related to blockchain applications management.
+ * Enables the component/hooks to use applications logic from one single place.
+ *
+ * @returns {Object} - Available blockchain applications array, a set
+ * application handler, get application by chain ID handler and  a delete
+ * application by ID handler.
+ */
+export function useBlockchainApplicationManagement() {
+  const dispatch = useDispatch();
+
+  const [currentApplication, setCurrentApplication] = useCurrentBlockchainApplication();
+
+  const { checkPinByChainId } = usePinBlockchainApplication();
+
+  const applicationsObject = useSelector(selectApplications);
+
+  const applications = useMemo(
+    () => {
+      const appsList = Object.values(applicationsObject);
+
+      return [...BLOCKCHAIN_APPLICATIONS_MOCK, ...appsList].map((app) => ({
+        ...app,
+        isPinned: checkPinByChainId(app.chainID),
+      })).sort((a) => (a.isPinned ? -1 : 1));
+    },
+    [applicationsObject, BLOCKCHAIN_APPLICATIONS_MOCK],
```

Comment:
```suggestion
    [applicationsObject, pins],
```
pin is the dependency
import file or function  like BLOCKCHAIN_APPLICATIONS_MOCK is not a dependency

Reply:
we should have this rule in eslint: `react-hooks/exhaustive-deps`. idk why we dont have it

Reply:
ok fixed now

Reply:
while pins is changed the memo usage sould be update other wise it dosnt show new state of pin